deltacat/aws/lakeformation/lakeformation_table_reader.py

```diff
@@ -0,0 +1,142 @@
+import logging
+import time
+import datetime
+import ray
+import pyarrow
+from typing import Optional, Callable, TypeVar
+from ray.data.impl.block_list import BlockList
+from ray.data.block import BlockMetadata
+from deltacat.aws.lakeformation.lakeformation_utils import ( 
+    get_lakeformation_client,
+    merge_blocklists
+)
+
+T = TypeVar('T')
+
+logger = logging.getLogger(__name__)
+
+@ray.remote
+def get_work_unit_result(region: str,
+                        query_id: str, work_unit_id: int, work_unit_token: str,
+                        mapper: Callable[[pyarrow.Table, str, str, dict], T],
+                        role_arn: str, **kwargs) -> T:
+
+    client = get_lakeformation_client(region, role_arn)
+    response = client.get_work_unit_results(
+        QueryId = query_id,
+        WorkUnitId = work_unit_id,
+        WorkUnitToken = work_unit_token
+    )
+
+    with pyarrow.ipc.open_stream(response["ResultStream"]._raw_stream) as reader:
+        downloaded_pyarrow_table = reader.read_all()
+
+    mapped_result = mapper(downloaded_pyarrow_table, query_id, work_unit_id, **kwargs)
+    return mapped_result
+
+class LakeformationTableReader:
```

Comment:
Can we implement storage interface?

Reply:
I initially thought about implementing the storage interface only but if you go through the storage interface it deals with streams deltas partitions none of which we will be using. Passing those variables around as null and sending the actual data inside a variable which doesnt make intuitive sense is something I wanted to avoid so I am not using that interface rather implementing this standalone.